source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/rtx_lidar/rtx_lidar.py

```diff
@@ -0,0 +1,346 @@
+                "outputElevation": False,
+                "outputNormal": False,
+                "outputVelocity": False,
+                "outputBeamId": False,
+                "outputEmitterId": False,
+                "outputMaterialId": False,
+                "outputObjectId": False,
+                "outputTimestamp": True,  # always turn on timestamp field
+            }
+
+            # create annotator node
+            annotator_type = "RtxSensorCpuIsaacCreateRTXLidarScanBuffer"
+            rep_annotator = rep.AnnotatorRegistry.get_annotator(annotator_type)
+            # turn on any optional data type returns
+            for name in self.cfg.optional_data_types:
+                if name == "azimuth":
+                    init_params["outputAzimuth"] = True
+                elif name == "elevation":
+                    init_params["outputElevation"] = True
+                elif name == "normal":
+                    init_params["outputNormal"] = True
+                elif name == "velocity":
+                    init_params["outputVelocity"] = True
+                elif name == "beamId":
+                    init_params["outputBeamId"] = True
+                elif name == "emitterId":
+                    init_params["outputEmitterId"] = True
+                elif name == "materialId":
+                    init_params["outputMaterialId"] = True
+                elif name == "objectId":
+                    init_params["outputObjectId"] = True
+
+            # transform the data output to be relative to sensor frame
+            if self.cfg.data_frame == "sensor":
+                init_params["transformPoints"] = False
+
+            rep_annotator.initialize(**init_params)
+
+            # Get render product
+            # From Isaac Sim 2023.1 onwards, render product is a HydraTexture so we need to extract the path
+            render_prod_path = rep.create.render_product(lidar_prim_path, [1, 1])
+            if not isinstance(render_prod_path, str):
+                render_prod_path = render_prod_path.path
+            self._render_product_paths.append(render_prod_path)
+
+            rep_annotator.attach(render_prod_path)
+            self._rep_registry.append(rep_annotator)
+
+            # Debug draw
+            if self.cfg.debug_vis:
+                self.writer = rep.writers.get("RtxLidarDebugDrawPointCloudBuffer")
+                self.writer.attach([render_prod_path])
+
+        # Create internal buffers
+        self._create_buffers()
+
+    def _create_buffers(self):
+        """Create buffers for storing data."""
+        # create the data object
+
+        # -- output data
+        # lazy allocation of data dictionary
+        # since the size of the output data is not known in advance, we leave it as None
+        # the memory will be allocated when the buffer() function is called for the first time.
+        self._data.output = TensorDict({}, batch_size=self._view.count, device=self.device)
```

Comment:
is TensorDict necessary here? we had a change that went in to remove TensorDict from the tiled camera class because it was causing performance issues

Reply:
Yeah I bet we can remove it.

Reply:
Hey I just checked both tiled_camera and camera use TensorDict for data.output. Did the change you mentioned go into the Nvidia internal fork?

Reply:
hmm I think it was updated in https://github.com/isaac-sim/IsaacLab/pull/1348